src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -3274,6 +3277,155 @@ def _if_stmt_handler(self, node, parent):
+        # construct or be an empty string if it is the default option.
+        type_string_name = parent.scope.symbol_table.next_available_name(
+            "type_string")
+        type_string_type = UnknownFortranType(
+            f"character(256) :: {type_string_name}\n")
+        type_string_symbol = DataSymbol(type_string_name, type_string_type)
+        parent.scope.symbol_table.add(type_string_symbol)
+
+        # Recreate the select type clause within a CodeBlock with
+        # the content of the clauses being replaced by a string
+        # capturing the name of the type or class clauses.
+        code = "program dummy\n"
+        code += f"select type({selector})\n"
+        for idx in range(select_idx+1):
+            if idx == default_idx:
+                continue
+            code += f"  {clause_type[idx]} ({guard_type[idx]})\n"
+            code += (f"    {type_string_name} = "
+                     f"\"{guard_type[idx].string.lower()}\"\n")
+        code += "end select\n"
+        code += "end program\n"
+        parser = ParserFactory().create(std="f2008")
+        reader = FortranStringReader(code)
+        fp2_program = parser(reader)
+        fp2_select_type = fp2_program.children[0].children[1]
+        code_block = CodeBlock(
+            [fp2_select_type], CodeBlock.Structure.STATEMENT, parent=parent)
+        # Handlers assume a single node is returned and in this
+        # implementation we create an assignment, a CodeBlock and an
+        # if statement (see later). Therefore we add the assignment
+        # and codeblock to the parent here and return the if statement
+        # in the usual way.
+        parent.addchild(Assignment.create(
+            Reference(type_string_symbol), Literal("", CHARACTER_TYPE)))
+        parent.addchild(code_block)
+
+        # Create the (potentially nested) if statement that replicates
+        # the functionality of the select type options (as select type
+        # is not supported directly in the PSyIR).
+        ifblock = None
+        currentparent = parent
+        for idx in range(select_idx+1):
+            if idx == default_idx:
+                continue
+            if ifblock:
+                # We already have an if so this is an else if
+                elsebody = Schedule(parent=currentparent)
+                currentparent.addchild(elsebody)
+                annotation = "was_select_type"
+                if clause_type[idx] == "CLASS IS":
+                    annotation = "was_class_type"
+                ifblock = IfBlock(annotations=[annotation])
+                elsebody.addchild(ifblock)
+            else:
+                annotation = "was_select_type"
+                if clause_type[idx] == "CLASS IS":
+                    annotation = "was_class_type"
+                ifblock = IfBlock(parent=currentparent,
+                                  annotations=[annotation])
+                stmt = ifblock
```

Comment:
pylint can't be sure that `stmt` is assigned to. Please could you rename it, e.g. 'outer_ifblock' and initialise it to None at L3386.

Reply:
Yes that also reads better, thanks.